### PR TITLE
Add fire station basecamp

### PR DESCRIPTION
**Pull request by curstwist**
#### Summary
 ```SUMMARY: Content "Adds fire station faction camp base option."```

#### Purpose of change
Begin making faction camps more varied.  

#### Describe the solution
Using the classic fire station for a test case.

Edit: at this point I think I've added all the base camp features except beds since they are linked to expansions.  I'm considering having more beds build-able, but it would be just as easy for the player to use other options, same goes for storage.

EDIT:
Added NPC quarters area for more bedrooms, common area.  Added skylights, and reduced chop shop to having a collection of tools/book.

Problems:
RESOLVED: Waiting on the spawning fix to get merged. 
The dismantle for chop shop and the farming don't seem to work in the core building.  I'm not sure if the radio is working either.

I've kept most constructions to the western area, and the back wall.  It assumes you didn't come in and smash up all the furniture already (which may be a mistake), and uses that furniture or gives you materials for any that is 'disappeared'.  

Finally, there is a fire truck in this base, but it blinks out of existence during update_mapgen.  You need to approach it for it to reappear, which I forgot to do for the screenshot.  Wonder woman would be envious.

#### Describe alternatives you've considered
Lots of other good buildings to choose from, but this was easily defensible and already has good space for the garage features.

#### Additional context
Posting this as a WIP/Draft to get some help troubleshooting initial spawn.

what it looks like finished:
firestation

The built out quarters.  I've only offered normal white walls for now.  The beds are full beds since mattresses should be common in the city.
quarters

### CODE REVIEW

**Comment by mlangsdorf** [comment]
It's awesome.

Some suggestions:
1. possibly add skylights to make it easier to craft with the garage doors closed.
2. move the garden 1 tile south and the radio tower 2 tiles south
3. so that there can be an option to put up a privacy fence around the garden, running east from the north side of the annex and south along the yellow pavement to the central brick pillar.
4. add pony walls on the east side to divide the vehicle bay into more housing instead of setting up a chop shop there - give the player the option to build either.
5. possibly add an option to board up the north window but maybe that isn't necessary.

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+      { "id": "foraging" },
+      { "id": "sorting" },
+      { "id": "logging" }
+    ],
+    "blueprint_requires": [ { "id": "not_an_upgrade" } ],
+    "blueprint_name": "basic survey"
+  },
+  {
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_kitchen1.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "result": "faction_base_modular_firestation1_kitchen1",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+    "description": "Now that we have some cover, we should reorganize that dining area into our kitchen.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "time": "1 h",
+    "construction_blueprint": "fbmc_firestation1_kitchen1_cleared",
+    "blueprint_name": "set up kitchen area.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "set up kitchen area",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+    "blueprint_requires": [ { "id": "fbmc_firestation1_0" } ],
+    "blueprint_provides": [ { "id": "fbmc_firestation1_kitchen" }, { "id": "tool_storage" }, { "id": "bed" } ],
+    "blueprint_excludes": [ { "id": "fbmc_firestation1_kitchen" } ],
+    "blueprint_autocalc": false
+  },
+  {
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_well.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "result": "faction_base_modular_firestation1_well",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+    "description": "We should build a well, put it near that winch. This will make our future here more secure.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_well",
+    "blueprint_name": "Add a well.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Add a well",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+    "blueprint_requires": [ { "id": "fbmc_firestation1_0" } ],
+    "blueprint_provides": [ { "id": "fbmc_firestation1_well" }, { "id": "relaying" }, { "id": "scouting" }, { "id": "patrolling" } ],
+    "blueprint_excludes": [ { "id": "fbmc_firestation1_well" } ],
+    "blueprint_autocalc": true
+  },
+  {
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_radio.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "result": "faction_base_modular_firestation1_radio",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+    "description": "Let's set up a radio tower to improve our recruitment efforts, we'll repurpose that terminal for it.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_radio_console",
+    "blueprint_name": "Add radio tower and console.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Add radio tower and console",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 7]
```diff
@@ -0,0 +1,466 @@
...
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_kitchen2_pantry",
+    "description": "Now that we have some cover, we should add some storage and a root cellar in our kitchen area.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "time": "2 h",
+    "construction_blueprint": "fbmc_firestation1_kitchen2_storage",
+    "blueprint_name": "add storage and root cellar to kitchen.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "add storage and root cellar to kitchen",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_farm",
+    "description": "Let's break up some pavement so we can farm.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "time": "24 h",
+    "construction_blueprint": "fbmc_firestation1_farm",
+    "blueprint_name": "Break up driveway for some planting.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Break up driveway for some planting",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 9]
```diff
@@ -0,0 +1,466 @@
...
+  {
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_garage1.",
+    "description": "Let's gather some tools so we can work on cars.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_garage_area",
+    "blueprint_name": "garage area.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "garage area",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 10]
```diff
@@ -0,0 +1,466 @@
...
+  {
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_living_area3",
+    "description": "Let's furnish the northwest bedroom.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_bedroom_nw",
+    "blueprint_name": "SW bedroom",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "NW bedroom",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 11]
```diff
@@ -0,0 +1,466 @@
...
+  {
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_living_area6",
+    "description": "Let's furnish the common area.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_quarters_common",
+    "blueprint_name": "Common area",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "common area",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 12]
```diff
@@ -0,0 +1,466 @@
...
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_pottery",
+    "description": "Let's build pottery kiln and get the tools we'll need for glassworking.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "skill_used": "fabrication",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_forge3",
+    "blueprint_name": "build a pottery kiln and glassworking tools.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "build a pottery kiln and glassworking tools",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 13]
```diff
@@ -0,0 +1,466 @@
...
+    "result": "faction_base_modular_firestation1_forgetools",
+    "description": "We need some tools for metal working.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "skill_used": "fabrication",
+    "autolearn": false,
+    "never_learn": true,
+    "time": "20 m",
+    "construction_blueprint": "fbmc_firestation1_forge4",
+    "blueprint_name": "Gather some advanced tools.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Gather some advanced tools",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 14]
```diff
@@ -0,0 +1,466 @@
...
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_workbench",
+    "description": "Let's get a nice workbench built to improve our work.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "skill_used": "fabrication",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_workbench",
+    "blueprint_name": "Build a workbench by the anvil.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Build a workbench by the anvil",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 15]
```diff
@@ -0,0 +1,466 @@
...
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_bellows",
+    "description": "Let's add a bellows and barrel for quenching the steel, we'll need more tools too.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "skill_used": "fabrication",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_bellows",
+    "blueprint_name": "Add bellows and barrel for quenching.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Add bellows and barrel for quenching",
```

File: data/json/recipes/basecamps/recipe_modular_firestation1.json [hunk 16]
```diff
@@ -0,0 +1,466 @@
...
+    "type": "recipe",
+    "result": "faction_base_modular_firestation1_drophammer",
+    "description": "Let's build a drop hammer for some mass production.",
+    "category": "CC_BUILDING",
+    "subcategory": "CSC_BUILDING_BASES",
+    "skill_used": "fabrication",
+    "autolearn": false,
+    "never_learn": true,
+    "construction_blueprint": "fbmc_firestation1_drophammer",
+    "blueprint_name": "Add a drop hammer.",
```
Comment (Night-Pryanik) [inline data/json/recipes/basecamps/recipe_modular_firestation1.json]:
```suggestion
    "blueprint_name": "Add a drop hammer",
```

**Comment by curstwist** [comment]
> 
> 
> Consistency. Please tidy up `blueprint_name` in all recipes - in some places it's full sentence, in other it's only just a name of location. Also consistency for upper case and lower case.

Fixed the periods.  I'll review their structure this evening.  Sometimes a fuller description is necessary but I try to avoid them due to the overall length of the name & the size of the space it displays.  I personally prefer the terse names.

**Comment by curstwist** [comment]
This PR is complete.

**Pull request merged**